enterprise_gateway/services/sessions/kernelsessionmanager.py

```diff
@@ -279,21 +291,47 @@ def persistence_root_default(self):
     def __init__(self, kernel_manager, **kwargs):
         super(FileKernelSessionManager, self).__init__(kernel_manager, **kwargs)
         if self.enable_persistence:
-            self.kernel_session_file = os.path.join(self._get_sessions_loc(), KERNEL_SESSIONS_FILE_NAME)
-            self.log.info("Kernel session persistence location: {}".format(self.kernel_session_file))
+            self.log.info("Kernel session persistence location: {}".format(self._get_sessions_loc()))
 
-    def save_sessions(self):
+    def delete_sessions(self):
         if self.enable_persistence:
-            with open(self.kernel_session_file, 'w') as fp:
-                json.dump(KernelSessionManager.pre_save_transformation(self._sessions), fp)
-                fp.close()
+            for kernel_id in self.sessions_to_remove:
```

Comment:
Per previous comment, I'd prefer to provide the list of sessions (kernel_ids) to remove rather than rely on the member variable.

Reply:
Done.